src/sentry/notifications/notifications/organization_request/base.py

```diff
@@ -36,7 +36,10 @@ def get_context(self) -> MutableMapping[str, Any]:
         return {}
 
     def determine_recipients(self) -> Iterable[RpcActor]:
-        return RpcActor.many_from_object(self.role_based_recipient_strategy.determine_recipients())
+        return [
+            RpcActor(id=user.id, actor_type=ActorType.USER)
```

Comment:
I wonder if you can can just invoke `self.role_based_recipient_strategy.determine_member_recipients()` to get the OrganizationMember objects and extract the `user_id` directly from there?  It would save a whole `user_service` call and other abstractions that aren't helping this logic.

Reply:
`OrganizationMemberMapping` => `OrganizationMemberMapping`

Reply:
In general, it begs the question if `role_based_recipient_strategy` on the whole can be refactored to use `OrganizationMemberMapping` and return the `RpcActor` objects in the same way (directly storing them and avoiding the `actor_id` field).  It would require getting more tests to pass but would have a broader positive impact.  Thoughts?

It's likely the code is organized this way now due to organize growth rather than thoughtful design.

Reply:
Agreed, that seems like a way better overall end state - I'll take a look

Reply:
I would be careful when you get around to changing how recipients are determined, there is also non-trivial logic buried in getsentry for billing notifications.

Reply:
After a bit more investigation, reworking things to use `OrganizationMemberMapping` looks like it'd be pretty complex. Since you all have way more experience: is that something that we can come back to or does it seem like it's worth taking the time to improve it now?

Reply:
> Since you all have way more experience: is that something that we can come back to or does it seem like it's worth taking the time to improve it now?

My preference is to come back to it. Notification settings storage and logic has a few bumpy patches and I think we'd be better off deferring a significant rework until we have more context and an end state that we want to end up with.